src/elements/timestamp.rs

```diff
@@ -198,219 +490,269 @@ impl Timestamp<'_> {
-                start,
-                end,
-                repeater: None,
-                delay: None,
-            },
-        ));
-    }
-
-    let (input, _) = space0(input)?;
-    // TODO: delay-or-repeater
-    let (input, _) = tag(">")(input)?;
-
-    if input.starts_with("--<") {
-        let (input, end) = parse_datetime(&input["--<".len()..])?;
-        let (input, _) = space0(input)?;
-        // TODO: delay-or-repeater
-        let (input, _) = tag(">")(input)?;
-        Ok((
-            input,
-            Timestamp::ActiveRange {
-                start,
-                end,
-                repeater: None,
-                delay: None,
-            },
-        ))
-    } else {
-        Ok((
-            input,
-            Timestamp::Active {
-                start,
-                repeater: None,
-                delay: None,
-            },
-        ))
+impl<'a> Parse<'a> for Timestamp<'a> {
+    fn parse(s: &'a str) -> Option<Timestamp<'a>> {
+        helper_for_parse_element(parse_timestamp(s))
     }
 }
 
-pub fn parse_inactive(input: &str) -> IResult<&str, Timestamp, ()> {
-    let (input, _) = tag("[")(input)?;
-    let (input, start) = parse_datetime(input)?;
-
-    if input.starts_with('-') {
-        let (input, (hour, minute)) = parse_time(&input[1..])?;
-        let (input, _) = space0(input)?;
-        // TODO: delay-or-repeater
-        let (input, _) = tag("]")(input)?;
-        let mut end = start.clone();
-        end.hour = Some(hour);
-        end.minute = Some(minute);
-        return Ok((
-            input,
-            Timestamp::InactiveRange {
-                start,
-                end,
-                repeater: None,
-                delay: None,
-            },
-        ));
-    }
+fn parse_time(input: &str) -> IResult<&str, (u8, u8), ()> {
+    let (input, hour) = map_res(take_while_m_n(1, 2, |c: char| c.is_ascii_digit()), |num| {
+        u8::from_str_radix(num, 10)
+    })(input)?;
+    let (input, _) = tag(":")(input)?;
+    let (input, minute) = map_res(take(2usize), |num| u8::from_str_radix(num, 10))(input)?;
+    Ok((input, (hour, minute)))
+}
 
-    let (input, _) = space0(input)?;
-    // TODO: delay-or-repeater
-    let (input, _) = tag("]")(input)?;
-
-    if input.starts_with("--[") {
-        let (input, end) = parse_datetime(&input["--[".len()..])?;
-        let (input, _) = space0(input)?;
-        // TODO: delay-or-repeater
-        let (input, _) = tag("]")(input)?;
-        Ok((
-            input,
-            Timestamp::InactiveRange {
-                start,
-                end,
-                repeater: None,
-                delay: None,
-            },
-        ))
-    } else {
-        Ok((
-            input,
-            Timestamp::Inactive {
-                start,
-                repeater: None,
-                delay: None,
-            },
-        ))
-    }
+fn parse_repeater_mark(input: &str) -> IResult<&str, RepeaterMark, ()> {
```

Comment:
You can use `value` here to simplify the parser:
```rust
fn parse_repeater_mark(input: &str) -> IResult<&str, RepeaterMark, ()> {
    alt((
        value(tag("++"), RepeaterMark::CatchUp),
        value(tag("+"), RepeaterMark::Cumulate),
        value(tag(".+"), RepeaterMark::Restart),
    ))(input)
}
```

Reply:
Nice, thanks